### PR TITLE
Move variables to config

### PR DESCRIPTION
**Pull request by ArneGudermann**
This draft move some vars from the utils in the config.
Beause we have some variables double set like:
https://github.com/viur-framework/viur-core/blob/198478b97d1fc9513b4bea4854164bd7e2ebb9ff/core/request.py#L167
and:
https://github.com/viur-framework/viur-core/blob/198478b97d1fc9513b4bea4854164bd7e2ebb9ff/core/utils.py#L32

we can combine this in the config and fill it when a instance start.

### CODE REVIEW

File: core/modules/file.py [hunk 1]
```diff
@@ -21,11 +21,11 @@
 from viur.core.prototypes.tree import SkelType, Tree, TreeSkel
 from viur.core.skeleton import skeletonByKind
 from viur.core.tasks import PeriodicTask, CallDeferred
-from viur.core.utils import projectID, sanitizeFileName
+from viur.core.utils import  sanitizeFileName
 
 credentials, project = google.auth.default()
 client = storage.Client(project, credentials)
-bucket = client.lookup_bucket("%s.appspot.com" % projectID)
+bucket = client.lookup_bucket("%s.appspot.com" % conf["viur.instance.project_id"])
```
Comment (phorward) [inline core/modules/file.py:28]: I love f-strings ;-)
```suggestion
bucket = client.lookup_bucket(f"""{conf["viur.instance.project_id"]}.appspot.com""")
```

File: core/modules/user.py [hunk 2]
```diff
@@ -683,7 +683,9 @@ def addSkel(self):
     def editSkel(self, *args, **kwargs):
         skel = super(User, self).editSkel().clone()
 
-        skel.password = PasswordBone(descr="Passwort", required=False)
+
+        skel.password.required=False
+
```
Comment (phorward) [inline core/modules/user.py:688]: This has nothing to do with the topic of the pull request!

File: core/modules/user.py [hunk 3]
```diff
@@ -860,7 +862,7 @@ def createNewUserIfNotExists():
                  userMod.validAuthenticationMethods])):  # It uses UserPassword login
         if not db.Query(userMod.addSkel().kindName).getEntry():  # There's currently no user in the database
             addSkel = skeletonByKind(userMod.addSkel().kindName)()  # Ensure we have the full skeleton
-            uname = "admin@%s.appspot.com" % utils.projectID
+            uname = "admin@%s.appspot.com" % conf["viur.instance.project_id"]
```
Comment (phorward) [inline core/modules/user.py:865]:
```suggestion
            uname = f"""admin@{conf["viur.instance.project_id"]}.appspot.com"""
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, thanks for making this pull request reality!

Anway, I still have some issues to be fixed. Please make use of the "apply suggestion" function on GitHub before manually applying changes, if you have questions on this, please contact me privately.

File: core/modules/user.py (hunk 2 above)
Comment (ArneGudermann) [reply core/modules/user.py:688]: This was for an other PR,but I forgot to delete it.

**Comment by ArneGudermann** [comment]
Now the version hash look like this:
form: ` conf["viur.instance.version_hash"]='2Gg7d3'`
to: ` conf["viur.instance.version_hash"]='5bfd8683ee'`

**Review by phorward — APPROVED** [review]
I'm fine with this.
What's with the other people?

**Comment by phorward** [comment]
@ArneGudermann Please turn this PR from draft to a real PR, so we can make a final review.

**Comment by phorward** [comment]
Oh yes, and please fix the available conflicts.

File: core/__init__.py [hunk 4]
```diff
@@ -220,8 +239,8 @@ def setup(modules: Union[object, ModuleType], render: Union[ModuleType, Dict] =
     # noinspection PyUnresolvedReferences
     import skeletons  # This import is not used here but _must_ remain to ensure that the
     # application's data models are explicitly imported at some place!
-    assert utils.projectID in conf["viur.validApplicationIDs"], \
-        "Refusing to start, applicationID %s is not in conf['viur.validApplicationIDs']" % utils.projectID
+    assert conf["viur.instance.project_id"] in conf["viur.validApplicationIDs"], \
+        f"""Refusing to start, {conf["viur.instance.project_id"]=} is not in {conf["viur.validApplicationIDs"]=}"""
```
Comment (sveneberth) [inline core/__init__.py:243]: As already described in https://github.com/viur-framework/viur-core/pull/507#discussion_r970617891 please replace these asserts with exceptions.

```suggestion
    if conf["viur.instance.project_id"] not in conf["viur.validApplicationIDs"]:
        raise RuntimeError(f"""Refusing to start, {conf["viur.instance.project_id"]=} is not in {conf["viur.validApplicationIDs"]=}""")
```

File: core/config.py [hunk 5]
```diff
@@ -178,4 +177,15 @@
 
     # Will be set to viur.core.version.__version__ in viur.core.__init__
     "viur.version": None,
+
+    # Will be set in viur.core.__init__
+    "viur.instance.is_dev_server": False,
```
Comment (sveneberth) [inline core/config.py:182]: Why shouldn't we set the value directly here?

```suggestion
    "viur.instance.is_dev_server": os.getenv('GAE_ENV') == "localdev",  
```
(import `os` before)

File: core/config.py [hunk 6]
```diff
@@ -178,4 +177,15 @@
...
+
+    "viur.instance.project_id": None,
+
+    # Name of this version as deployed to the appengine will be also set in viur.core.__init__
+    "viur.instance.app_version": None,
```
Comment (sveneberth) [inline core/config.py:187]:
```suggestion
    "viur.instance.app_version": os.getenv("GAE_VERSION"),
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by phorward — DISMISSED** [review]

**Review by XeoN-GHMB — APPROVED** [review]

File: core/request.py [hunk 7]
```diff
@@ -105,6 +105,16 @@ def __init__(self, request: webob.Request, response: webob.Response):
...
 
+    def __getattr__(self, attr):
+        if attr == "isDevServer":
+            import warnings
+            msg = f"Use of `{attr}` is deprecated; Use conf[\"viur.project.is_dev_server\"] instead!"
+            warnings.warn(msg, DeprecationWarning, stacklevel=2)
+            logging.warning(msg)
+            return conf["viur.project.is_dev_server"]
+
+        return super().__getattr__(attr)
```
Comment (sveneberth) [inline core/request.py:116]: We have properties for this in python!

```suggestion
    @property
    def isDevServer(self) -> bool:
          import warnings
          msg = f"Use of `{attr}` is deprecated; Use conf[\"viur.project.is_dev_server\"] instead!"
          warnings.warn(msg, DeprecationWarning, stacklevel=2)
          logging.warning(msg)
          return conf["viur.project.is_dev_server"]
```

File: core/utils.py [hunk 8]
```diff
@@ -287,4 +279,19 @@ def normalizeKey(key: Union[None, 'db.KeyClass']) -> Union[None, 'db.KeyClass']:
     return db.Key(key.kind, key.id_or_name, parent=parent)
 
 
-from viur.core.skeleton import SkeletonInstance
```
Comment (sveneberth) [inline core/utils.py:290]: Please re-add the `SkeletonInstance` import.

It's used here:
https://github.com/viur-framework/viur-core/blob/c3d5dff409535a802d336f237410f2a0133d4a3f/core/utils.py#L180
and for type hints.

File: core/logging.py [hunk 9]
```diff
@@ -35,9 +36,10 @@ def emit(self, record: logging.LogRecord):
             message,
             resource=self.resource,
             labels={
-                "project_id": projectID,
+                "project_id": conf["viur.instance.project_id"],
                 "module_id": "default",
-                "version_id": appVersion if not isLocalDevelopmentServer else "dev_appserver",
+                "version_id": conf["viur.instance.app_version"] if not conf[
+                    "viur.instance.is_dev_server"] else "dev_appserver",
```
Comment (sveneberth) [inline core/logging.py:42]: This line break is very ugly.


I suggest sth. like this:
```suggestion
                "version_id": (conf["viur.instance.app_version"] 
                                         if not conf["viur.instance.is_dev_server"] 
                                         else "dev_appserver"),
```

Same for lines 15-16 above.

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: core/utils.py (hunk 8 above)
Comment (phorward) [reply core/utils.py:290]: aha, so why is it defined at the end of the file, and not inline? And even without any comment or hint.

Comment (phorward) [reply core/utils.py:290]: I've resolved this case inline in 2fa27d9

Comment (phorward) [reply core/utils.py:290]: Let's do it in another PR, separated from this one.

File: core/__init__.py [hunk 10]
```diff
@@ -24,19 +24,14 @@
...
 import webob
-
-# Copy our Version into the config so that our renders can access it
-conf["viur.version"] = tuple(__version__.split(".", 3))
+from types import ModuleType
+from typing import Callable, Dict, Union
+from viur.core import session, errors, i18n, request, utils
+from viur.core.config import conf
+from viur.core.tasks import TaskHandler, runStartupTasks
+from viur.core import logging as viurLogging  # Initialize request logging # noqa: E402
```
Comment (sveneberth) [inline core/__init__.py:34]: It's now at the top of the file, so it should fit with E402 :)
```suggestion
from viur.core import logging as viurLogging  # Initialize request logging
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Pull request merged**